stl/inc/expected

```diff
@@ -0,0 +1,1020 @@
+        requires is_constructible_v<_Err, _UErr>
+    constexpr expected(unexpected<_UErr>&& _Other) noexcept(is_nothrow_constructible_v<_Err, _UErr>) /* strengthened */
+        : _Has_value(false), _Unexpected(_STD forward<_UErr>(_Other._Unexpected)) {}
+
+    template <class... _Args>
+        requires is_constructible_v<_Ty, _Args...>
+    constexpr explicit expected(in_place_t, _Args&&... _Vals) noexcept(
+        is_nothrow_constructible_v<_Ty, _Args...>) /* strengthened */
+        : _Has_value(true), _Value(_STD forward<_Args>(_Vals)...) {}
+
+    // clang-format off
+    template <class _Uty, class... _Args>
+        requires is_constructible_v<_Ty, initializer_list<_Uty>&,_Args... >
+    constexpr explicit expected(in_place_t, initializer_list<_Uty> _Ilist, _Args&&... _Vals) noexcept(
+        is_nothrow_constructible_v<_Ty, initializer_list<_Uty>&, _Args...>) /* strengthened */
+        : _Has_value(true), _Value(_Ilist, _STD forward<_Args>(_Vals)...) {}
+    // clang-format on
+
+    template <class... _Args>
+        requires is_constructible_v<_Err, _Args...>
+    constexpr explicit expected(unexpect_t, _Args&&... _Vals) noexcept(
+        is_nothrow_constructible_v<_Err, _Args...>) /* strengthened */
+        : _Has_value(false), _Unexpected(_STD forward<_Args>(_Vals)...) {}
+
+    // clang-format off
+    template <class _Uty, class... _Args>
+        requires is_constructible_v<_Err, initializer_list<_Uty>&, _Args... >
+    constexpr explicit expected(unexpect_t, initializer_list<_Uty> _Ilist, _Args&&... _Vals) noexcept(
+        is_nothrow_constructible_v<_Err, initializer_list<_Uty>&, _Args...>) /* strengthened */
+        : _Has_value(false), _Unexpected(_Ilist, _STD forward<_Args>(_Vals)...) {}
+    // clang-format on
+
+    // [expected.object.dtor]
+    constexpr ~expected() noexcept {
+        if (_Has_value) {
+            _Value.~_Ty();
+        } else {
+            _Unexpected.~_Err();
+        }
+    }
+
+    // clang-format off
+    ~expected() requires is_trivially_destructible_v<_Ty> && is_trivially_destructible_v<_Err> = default;
+    // clang-format on
+
+
+    // [expected.object.assign]
+    template <class _First, class _Second, class... _Args>
+    static constexpr void _Reinit_expected(_First& _New_val, _Second& _Old_val, _Args&&... _Vals) noexcept(
+        is_nothrow_constructible_v<_First, _Args...> || is_nothrow_move_constructible_v<_First>) // strengthened
+    {
+        if constexpr (is_nothrow_constructible_v<_First, _Args...>) {
+            _STD destroy_at(_STD addressof(_Old_val));
+            _STD construct_at(_STD addressof(_New_val), _STD forward<_Args>(_Vals)...);
+        } else if constexpr (is_nothrow_move_constructible_v<_First>) {
+            _First _Tmp(_STD forward<_Args>(_Vals)...);
+            _STD destroy_at(_STD addressof(_Old_val));
+            _STD construct_at(_STD addressof(_New_val), _STD move(_Tmp));
+        } else {
+            _Second _Tmp(_STD move(_Old_val));

```

Comment:
I think it's useful to note that `_Second` is nothrow move constructible in this branch (because all functions that call `_Reinit_expected` are appropriately constrained).

```suggestion
            _STL_INTERNAL_STATIC_ASSERT(is_nothrow_move_constructible_v<_Second>);
            _Second _Tmp(_STD move(_Old_val));
```